src/components/plots/barPlot.ts

```diff
@@ -165,30 +288,20 @@ export module Plot {
         return d3.select();
       }
 
-      var xExtent: Extent = this._parseExtent(xValOrExtent);
-      var yExtent: Extent = this._parseExtent(yValOrExtent);
-
       // currently, linear scan the bars. If inversion is implemented on non-numeric scales we might be able to do better.
       var bars = this._datasetKeysInOrder.reduce((bars: any[], key: string) =>
-        bars.concat(this._getBarsFromDataset(key, xExtent, yExtent))
+        bars.concat(this._getBarsFromDataset(key, xValOrExtent, yValOrExtent))
       , []);
 
       return d3.selectAll(bars);
     }
 
-    private _getBarsFromDataset(key: string, xExtent: Extent, yExtent: Extent): any[] {
-      // the SVGRects are positioned with sub-pixel accuracy (the default unit
-      // for the x, y, height & width attributes), but user selections (e.g. via
-      // mouse events) usually have pixel accuracy. A tolerance of half-a-pixel
-      // seems appropriate:
-      var tolerance: number = 0.5;
+    private _getBarsFromDataset(key: string, xValOrExtent: number | Extent, yValOrExtent: number | Extent): any[] {
       var bars: any[] = [];
 
       var drawer = <_Drawer.Element>this._key2PlotDatasetKey.get(key).drawer;
       drawer._getRenderArea().selectAll("rect").each(function(d) {
-        var bbox = this.getBBox();
-        if (bbox.x + bbox.width >= xExtent.min - tolerance && bbox.x <= xExtent.max + tolerance &&
-            bbox.y + bbox.height >= yExtent.min - tolerance && bbox.y <= yExtent.max + tolerance) {
+        if (Bar._intersectsBBox(xValOrExtent, yValOrExtent, this.getBBox())) {
```

Comment:
I would very much like to keep the idea of only using extents in development code